packages/financial-templates-lib/src/price-feed/ExpressionPriceFeed.js

```diff
@@ -0,0 +1,114 @@
+const assert = require("assert");
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const Web3 = require("web3");
+const { create, all } = require("mathjs");
+const math = create(all, { number: "BigNumber", precision: 100 });
+
+// Gets balancer spot and historical prices. This price feed assumes that it is returning
+// prices as 18 decimals of precision, so it will scale up the pool's price as reported by Balancer contracts
+// if the user specifies that the Balancer contract is returning non-18 decimal precision prices.
+class ExpressionPriceFeed extends PriceFeedInterface {
```

Comment:
Another meta comment: If you wanted a price feed that is the result of two existing, approved price feeds run through this calculator, does that price feed need to be approved through the UMIP process, or would people be free to do price feed mash-ups in new synthetics without an extra vote?

Reply:
@pemulis this is a great question. To put it simply, the latter. However, I want to make a larger point. None of our price feeds know anything about approved identifiers. They only know about the default configs in the price feed directory. Those default configs allow us to automatically construct price feeds based on the identifier the EMP is using without needing a custom config input. When an UMIP is approved to add an identifier, we generally add a default config for it.

In addition to expressions automatically being able to use default configs with no additional information, you will also be able to specify a `customFeeds` map inside the ExpressionPriceFeed config that maps your own custom symbols so you're not limited to the default configs. Here's an example of a working config:
```js
{
  type: "expression",
  expression: "ETHBTC + MYCUSTOMFEED",
  customFeeds: {
    MYCUSTOMFEED: {
      type: "cryptowatch",
      exchange: "coinbase-pro",
      pair: "ethusd"
    }
  }
}
```

Reply:
The only thing actually on-chain is the identifier string, right?

Reply:
Yep!